packages/data-stores/src/auth/controls.ts

```diff
@@ -0,0 +1,39 @@
+/**
+ * External dependencies
+ */
+import { createRegistryControl } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { wpcomRequest, WpcomClientCredentials } from '../utils';
+import { fetchAuthOptions, wpLogin } from './actions';
+import { STORE_KEY } from './constants';
+
+export function createControls( clientCreds: WpcomClientCredentials ) {
+	return {
+		SELECT_USERNAME_OR_EMAIL: createRegistryControl( registry => () => {
+			return registry.select( STORE_KEY ).getUsernameOrEmail();
+		} ),
+		FETCH_AUTH_OPTIONS: async ( action: ReturnType< typeof fetchAuthOptions > ) => {
+			// TODO url encode so that this works with email
+			const { usernameOrEmail } = action;
+
+			return await wpcomRequest( {
+				path: `/users/${ usernameOrEmail }/auth_options`,
+				apiVersion: '1.1',
+			} );
+		},
+		WP_LOGIN: async ( { action, params }: ReturnType< typeof wpLogin > ) => {
+			return await wpcomRequest( {
+				path: '/wp-login.php?action=' + action,
```

Comment:
Since we only call `WP_LOGIN` from the one place, I would consider in-lining the action so it hits `/wp-login.php?action=login-endpoint` every time.

Reply:
I was thinking of re-using the `WP_LOGIN` control for the `wp-login.php?action=two-step-authentication-endpoint` request too, which is why I didn't hard code the query param.

But maybe re-using the same control for both cases will be too messy 🤔 I might make seperate `LOGIN_ENDPOINT` and `TWO_STEP_AUTHENTICATION_ENDPOINT` controls but I'm not sure yet.

Reply:
yea up to you, I think passing the param into a URL always makes me a bit uncomfortable because of things like `?action=` +`action&param2=something`. It's harmless if that does occur because `wp-login.php` is a public endpoint anyway... and we would be hardcoding `action` to an action name string anyway... it just makes me feel slight discomfort for some reason :joy:

Reply:
Excellent point! Should have thought more of the evils of string concat!

I'll see if there's a real url builder library already in the `package.json`, but a seperate control might just be simpler.

Reply:
Maybe `encodeURIComponent()` is the best way to go... or if you can have a type on action so it can only be `two-step-authentication-endpoint | login-endpoint`. Yea but it's not necessary for code/security just for OCD satisfaction.

Reply:
Now I've seen it I can't unsee it lol